src/mongoose_iq_handler.erl

```diff
@@ -0,0 +1,133 @@
+
+-module(mongoose_iq_handler).
+-include("mongoose.hrl").
+
+%%----------------------------------------------------------------------
+%% Types
+%%----------------------------------------------------------------------
+
+-record(iq_handler, {iq_handler_fn :: iq_handler(),
+                     extra :: map(),
+                     execution_method :: execution_method()}).
+
+-type t() :: #iq_handler{}.
```

Comment:
This one is a common pattern, when the module name already says everything about the type, and the type is exported and referenced as `mongoose_iq_handler:t()`, so I'm kinda ok with this convention 🤔

Reply:
Fully agree with @NelsonVides, it's a common pattern for the modules that suppose to be a wrapper for just one structure.

Reply:
If I was to guess what the `t` means, I would fail, especially that there is another type named `iq_handler`. For me it is a bad convention anyway, type names should be descriptive, we don't call functions 'f', variables 'v' and records 'r' for the very same reason. Let's not break one window, because others will follow.

Reply:
I don't see how the type `mongoose_iq_handler:t()` is better than `mongoose_iq_handler:iq_handler()`.

But you are absolutely right, the type that is currently called `iq_handler()` must be renamed into `iq_handler_fn()` to avoid collisions with `#iq_handler{}` record.